test/e2e/framework/pv/pv.go

```diff
@@ -873,3 +879,49 @@ func GetDefaultFSType() string {
 	}
 	return "ext4"
 }
+
+// returns the patch to apply to make a storage class the default
+// one or to strip it from being the default one
+func getDefaultStorageClassPatch(isDefault bool) []byte {
+	return []byte(fmt.Sprintf(`{"metadata": {"annotations": {"%s": "%s"}}}`, isDefaultStorageClassAnnotation, strconv.FormatBool(isDefault)))
+}
+
+// SetAsDefaultStorageClass sets an input StorageClass object to be the default storage class.
+// There are two scenarios:
+// - there's a default storage class, in this case the annotations to make it default are removed
+//   and added to the input storage class
+// - there's no default storage class, just adds the annotations to the input storage class
+// NOTE: the wanted StorageClass should be already in the cluster, this method will just make
+// sure it's the default StorageClass
+func SetAsDefaultStorageClass(client clientset.Interface, sc *storagev1.StorageClass) {
```

Comment:
for tests that run in parallel, will it cause issues for other tests using the existing default storageclass?

Reply:
yes, this assumes that only this test will be run, I can add the `[Serial]` annotation

Reply:
this will only used by one test?

Reply:
@msau42 could you also help review?

Reply:
I don't think we should be modifying cluster-wide settings in our tests. How much overlap do these tests have with testsuites/multivolume.go? Can we migrate/copy those tests there instead?

Reply:
I see, so the idea is to make the test work with a CSI driver only, what we can do is skip if the node os distro is windows on the intree tests and also make a copy of the test in multivolume.go that works with windows nodes too, how does that sound?

Reply:
Tests under testsuites also work for in-tree if you configure it.

Reply:
https://github.com/kubernetes/kubernetes/blob/fa558043c02cae10ffbe001b46a7166231d07f60/test/e2e/storage/drivers/in_tree.go

Reply:
got it, I'll dig a little bit and I'll add a comment with the plan for this test

Reply:
@msau42 I remembered you mentioned something about `testsuites` not having tests for the default StorageClass in https://github.com/kubernetes/kubernetes/pull/99346#discussion_r581442391, the objective of this test is to test the default StorageClass which is the intree one and not the PD CSI one (correct me on this if I'm wrong), my cluster has this one created by default:

```
allowVolumeExpansion: true
 apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   annotations:
     storageclass.kubernetes.io/is-default-class: "true"
   creationTimestamp: "2021-03-17T17:41:27Z"
   labels:
     addonmanager.kubernetes.io/mode: EnsureExists
   name: standard
   resourceVersion: "5105323"
   uid: 9b6ca4e7-80b3-4164-b5a7-53ba4af085a1
 parameters:
   type: pd-standard
 provisioner: kubernetes.io/gce-pd
 reclaimPolicy: Delete
 volumeBindingMode: Immediate
```

I agree that we shouldn't modify the cluster state, would an alternative solution be to have the PD CSI StorageClass bootstrapped as the default StorageClass when the cluster is created instead?

Reply:
Is it important that we actually test the default storageclass? Or can we test this functionality with a non-default storageclass?

Reply:
We can definitely test it with a non-default StorageClass, the test is quite simple and I believe it might be already a testcase in multivolumes.go which is much more comprehensive (test in a single node, test in multiple nodes, test with different volume modes, etc) however I thought that the intention of this test was to test the provisioning of volumes with the default StorageClass

After checking the file `testsuites/multivolume.go` I understood that for every volume created there's an accompanying storage class i.e. if `numVols` is the number of volumes that we want to provision we'll also create `numVols` StorageClasses

```go
		for i := 0; i < numVols; i++ {
			testVolumeSizeRange := t.GetTestSuiteInfo().SupportedSizeRange
            // creates the StorageClass
			resource := storageframework.CreateVolumeResource(driver, l.config, pattern, testVolumeSizeRange)
			l.resources = append(l.resources, resource)
			pvcs = append(pvcs, resource.Pvc)
		}
```

Moving/copying this test to multivolume.go might not be that useful, I can try making some of the multivolume.go tests work with windows nodes although I haven't found equivalent commands for `test -f` and `test -b` given that Windows only creates symlinks as @jingxu97 kindly explained, what if I skip this test in windows and try to find equivalent commands for windows in multivolume.go instead?

Reply:
@msau42 @jingxu97 I've disabled this intree test with windows nodes and instead I worked on enabling the testsuites/multivolume tests with windows nodes: https://github.com/kubernetes/kubernetes/pull/100641